rust/arrow/src/buffer.rs

```diff
@@ -213,13 +222,33 @@ impl<T: AsRef<[u8]>> From<T> for Buffer {
         let len = slice.len() * mem::size_of::<u8>();
         let capacity = bit_util::round_upto_multiple_of_64(len);
         let buffer = memory::allocate_aligned(capacity);
+        // JUSTIFICATION
+        //  Benefit
+        //      It is often useful to create a buffer from bytes, typically when they are allocated by external sources
+        //  Soundness
+        //      * The pointers are non-null by construction
+        //      * alignment asserted above
+        //  Unsoundness
+        //      * There is no guarantee that the memory regions do are non-overalling, but `memcpy` requires this.
```

Comment:
Since `buffer` is freshly allocated it should not be possible for the memory regions to overlap

Reply:
Good point. This was copied from a previous version and it slipped. :/